### ISSUE TITLE
[BUG] Can't import specifically charater

### ISSUE THREAD

**Issue by huybin1234b** [SillyTavern/SillyTavern#2867]
### Environment

🪟 Windows

### System

nodejs ver v20.11.0,Windows 11 23H2 (OS Build 22631.4169),Thorium browser AVX2 (Idk what version I use)

### Version

1.12.5 staging commit 0b0bd273

### Desktop Information

****

### Describe the problem

when I import [this](https://janitorai.com/characters/a9a42756-9db2-4203-80da-c5055f06a9ed_character-emi-%F0%9F%92%9A) charater my ST don't import and it get an error like the log bellow

### Additional info

Terminal Log
![image](https://github.com/user-attachments/assets/b1b3be26-3b9a-4cd9-9a6f-d09f3e6ed631)

### Please tick the boxes

- [X] I have explained the issue clearly, and I included all relevant info
- [X] I have checked that this [issue hasn't already been raised](https://github.com/SillyTavern/SillyTavern/issues?q=is%3Aissue)
- [X] I have checked the [docs](https://docs.sillytavern.app/) ![important](https://img.shields.io/badge/Important!-F6094E)

**Comment by Wolfsblvt** [comment]
Can confirm. That is reproducable for me.

Likely an issue with special characters in the filename not being sanitized away correctly.

**Comment by Cohee1207** [comment]
I couldn't reproduce on staging.
Tried Windows and Mac.

**Comment by Wolfsblvt** [comment]
Maybe this is a specific node version issue?
Because It somewhat looks like the issue is happening inside `write-file-atomic` package, when we make this call here.
https://github.com/SillyTavern/SillyTavern/blob/83c3f6d1bfcb240eaeda4c0f109160352de91175/src/endpoints/characters.js#L78-L82

Like I shared on discord, I can reproduce this, and can track it down to *inside* the node module, where it tries to open the filestream, or something.
![image](https://github.com/user-attachments/assets/e55eb3d9-687a-4d91-986a-ac569bfcb6cf)

Dunno what to do next though.

My node is **v21.1.0**, so different to OP.

**Comment by LenAnderson** [comment]
You don't even need to go through ST to test this.

`test.mjs` (run directly with node)
```javascript
import { createRequire } from 'module';
const require  = createRequire(import.meta.url);
const fs = require('fs');

let tmpfile = 'Emi 💚.png';
const options = {};
try {
	fs.openSync(tmpfile, 'w', options.mode || 0o666)
} catch (ex) {
	console.log(ex);
}
```

Probably this known issue with unicode surrogate pairs in FS.
https://github.com/nodejs/node/issues/48673

Should be fixed in node 20.12.0.

**Comment by Wolfsblvt** [comment]
> Should be fixed in node 20.12.0.

Whelp, seems like I have to either downgrade my node then or go to a much newer version.
Don't ask me why I have 21.1.0 installed.

**Comment by Cohee1207** [comment]
Confirmed as not an ST bug.

Remediation proposed: update to the newest version of Node LTS (v20.17.0 as of now, but at least 20.12.0).
